src/spi/http/curlHttpClient.ts

```diff
@@ -0,0 +1,108 @@
+import * as _ from "lodash";
+import * as os from "os";
+import { logger } from "../..";
+import { doWithRetry } from "../../util/retry";
+import {
+    spawnAndWatch,
+    SuccessIsReturn0ErrorFinder,
+    WritableLog,
+} from "../../util/spawned";
+import {
+    DefaultHttpClientOptions,
+    HttpClient,
+    HttpClientFactory,
+    HttpClientOptions,
+    HttpResponse,
+} from "./httpClient";
+
+/**
+ * Curl based HttpClient implementation.
+ */
+export class CurlHttpClient implements HttpClient {
+
+    public exchange<T>(url: string,
+                       options: HttpClientOptions = {}): Promise<HttpResponse<T>> {
+
+        const optionsToUse: HttpClientOptions = {
+            ...DefaultHttpClientOptions,
+            ...options,
+        };
+
+        // Prepare headers
+        const headers = [];
+        _.forEach(optionsToUse.headers, (v, k) => {
```

Comment:
Is there a reason you use `_.forEach`  and `push` instead of `_.map`?

Reply:
`map` only works on arrays not on object literals.

Reply:
`_.map` should work on any collection https://lodash.com/docs/#map